testing/test_transforms.py

```diff
@@ -0,0 +1,353 @@
+
+    # Transform on Numpy
+    transform = NormalizeInstance()
+    do_im, _ = transform(im.copy(), metadata_in)
+    # Check normalization
+    for i in do_im:
+        assert abs(np.mean(i) - 0.0) <= 1e-2
+        assert abs(np.std(i) - 1.0) <= 1e-2
+
+    # Transform on Tensor
+    tensor, metadata_tensor = NumpyToTensor()(im, metadata_in)
+    do_tensor, _ = transform(tensor, metadata_tensor)
+    # Check normalization
+    for t in do_tensor:
+        assert abs(t.mean() - 0.0) <= 1e-2
+        assert abs(t.std() - 1.0) <= 1e-2
+
+
+def _test_Crop(im_seg, crop_transform):
+    im, seg = im_seg
+    metadata_ = {'data_shape': im[0].shape}
+    metadata_in = [metadata_ for _ in im] if isinstance(im, list) else {}
+
+    if crop_transform.__class__.__name__ == "ROICrop":
+        _, metadata_in = crop_transform(seg, metadata_in)
+        for metadata in metadata_in:
+            assert "crop_params" in metadata
+
+    # Apply transform
+    crop_transfrom_size = crop_transform.size if not crop_transform.is_2D else crop_transform.size[:2]
+    do_im, do_metadata = crop_transform(im, metadata_in)
+    do_seg, do_seg_metadata = crop_transform(seg, metadata_in)
+
+    # Loop and check
+    for idx, i in enumerate(im):
+        # Check data shape
+        assert do_im[idx].shape == crop_transfrom_size
+        assert do_seg[idx].shape == crop_transfrom_size
+        # Check metadata
+        assert do_metadata[idx]['crop_params'] == do_seg_metadata[idx]['crop_params']
+
+    # Apply undo transform
+    undo_im, _ = crop_transform.undo_transform(do_im, do_metadata)
+    undo_seg, _ = crop_transform.undo_transform(do_seg, do_seg_metadata)
+
+    # Loop and check
+    for idx, i in enumerate(im):
+        # Check data shape
+        assert undo_im[idx].shape == i.shape
+        assert undo_seg[idx].shape == seg[idx].shape
+        # Check data type
+        assert do_im[idx].dtype == undo_im[idx].dtype == i.dtype
+        assert do_seg[idx].dtype == undo_seg[idx].dtype == seg[idx].dtype
+        # Check data consistency
+        fh, fw, fd, _, _, _ = do_metadata[idx]['crop_params']
+        th, tw, td = crop_transform.size
+        if crop_transform.is_2D:
+            assert np.array_equal(i[fh:fh+th, fw:fw+tw], undo_im[idx][fh:fh+th, fw:fw+tw])
+            assert np.array_equal(seg[idx][fh:fh+th, fw:fw+tw], undo_seg[idx][fh:fh+th, fw:fw+tw])
+            # Plot for debugging
+            if DEBUGGING:
+                plot_transformed_sample(seg[idx], undo_seg[idx], ['raw', 'undo'])
+                plot_transformed_sample(i, undo_im[idx], ['raw', 'undo'])
+        else:
+            assert np.array_equal(i[fh:fh+th, fw:fw+tw, fd:fd+td], undo_im[idx][fh:fh+th, fw:fw+tw, fd:fd+td])
+            assert np.array_equal(seg[idx][fh:fh+th, fw:fw+tw, fd:fd+td], undo_seg[idx][fh:fh+th, fw:fw+tw, fd:fd+td])
+
+
+@pytest.mark.parametrize('im_seg', [create_test_image(100, 100, 0, 2)])
+@pytest.mark.parametrize('crop_transform', [CenterCrop((80, 60)),
```

Comment:
The test were passing though, but happy to change that

Reply:
weird, I had an error, that's why I put the comment, but tuples are not mutable so it makes sense to put a list since if it's 2D we modify  the crop dimensions :)

Reply:
done